### PR TITLE
pgcraft: use hwmap to scope specific samplers code paths

### PR DESCRIPTION
**Pull request by mbouron-gpsw**

### CODE REVIEW

File: libnodegl/pgcraft.c [hunk 1]
```diff
@@ -268,9 +268,10 @@ static int prepare_texture_info_fields(struct pgcraft *s, const struct pgcraft_p
     for (int i = 0; i < NGLI_INFO_FIELD_NB; i++) {
         struct pgcraft_texture_info_field *field = &info->fields[i];
 
-        field->type = types_map[i];
-        if (field->type == NGLI_TYPE_NONE)
+        const int type = types_map[i];
+        if (type == NGLI_TYPE_NONE)
             continue;
+        field->type = type;
```
Comment (cboesch-gpsw) [inline libnodegl/pgcraft.c:291]: So previously `field->type` was set, even when it was `NGLI_TYPE_NONE`, which is not the case anymore. I'm assuming this is not an issue?

**Review by cboesch-gpsw — COMMENTED** [review]
Not a huge fan of having the support check and the generated code so distant (due to the risk of desync between the 2 / mistake), but let's say it's fine for now, until that code is somehow moved to their respective backends.

File: libnodegl/pgcraft.c (hunk 1 above)
Comment (mbouron-gpsw) [reply libnodegl/pgcraft.c:291]: It should be ok as the info structure is 0-initialized according the current code workflow:
```c
        struct pgcraft_texture_info info = {
            .stage     = texture->stage,
            .precision = texture->precision,
            .texture   = texture->texture,
            .image     = texture->image,
            .format    = texture->format,
            .writable  = texture->writable,
        };

        int ret = prepare_texture_info_fields(s, params, graphics, texture, &info);
        if (ret < 0)
            return ret;

```

Comment (cboesch-gpsw) [reply libnodegl/pgcraft.c:291]: Thanks

**Review by cboesch-gpsw — APPROVED** [review]

**Comment by cboesch-gpsw** [comment]
See https://github.com/gopro/gopro-lib-node.gl/pull/388 as an alternative.

**Comment by mbouron-gpsw** [comment]
> See #388 as an alternative.

Rebased on top of your PR.

**Pull request merged**